File: demos/showcase/android/compose/src/main/java/com/bajutsu/showcase/compose/NoticesScreen.kt
package com.bajutsu.showcase.compose

import androidx.activity.compose.BackHandler
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

// Tab: Notices (SPEC §5.5). A plain vertical list of 20 static notices — intentionally longer than one
// screen, so the bottom rows start off-screen and reaching notice.row.20 is the canonical
// scroll-to-element target. Tapping a row pushes its detail.
@Composable
fun NoticesScreen(model: AppModel) {
    val pushedId = model.noticesPath.lastOrNull()
    if (pushedId == null) {
        NoticesList(model)
    } else {
        BackHandler { model.noticesPath.removeAt(model.noticesPath.lastIndex) }
        NoticeDetailScreen(model, pushedId)
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun NoticesList(model: AppModel) {
    Column(Modifier.fillMaxSize()) {
        TopAppBar(title = { Text("Notices") })
        LazyColumn(Modifier.weight(1f)) {
            items(model.notices, key = { it.id }) { notice ->
                Text(
                    notice.title,
                    Modifier
                        .fillMaxWidth()
                        .clickable { model.noticesPath.add(notice.id) }
                        .padding(16.dp)
                        .aid("notice.row.${notice.id}"),
                )
                HorizontalDivider()
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun NoticeDetailScreen(model: AppModel, id: Int) {
    val notice = model.notice(id)
    Column(Modifier.fillMaxSize()) {
        TopAppBar(title = { Text(notice?.title ?: "Notice $id") })
        Column(Modifier.padding(16.dp)) {
            Text(notice?.title ?: "Notice $id", Modifier.aid("notice.detail.title"), style = MaterialTheme.typography.titleLarge)
            Text(notice?.body ?: "", Modifier.aid("notice.detail.body"), color = MaterialTheme.colorScheme.onSurfaceVariant)
        }
    }
}

File: demos/showcase/flutter/lib/screens/notices_screen.dart
import 'package:flutter/material.dart';

import '../accessibility.dart';
import '../model.dart';
import 'common.dart';

/// Tab: Notices (SPEC §5.5). A plain vertical list of 20 static notices — intentionally longer than
/// one screen, so the bottom rows start off-screen and reaching `notice.row.20` is the canonical
/// scroll-to-element target. Because a `ListView` builds its children lazily, an off-screen row is
/// not in the accessibility tree until scrolled into view — the culling behavior BE-0008 verifies.
/// Tapping a row pushes its detail on the root navigator.
class NoticesScreen extends StatelessWidget {
  const NoticesScreen({super.key, required this.model});

  final AppModel model;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Notices')),
      body: ListView(
        children: [
          for (final notice in model.notices)
            aid(
              'notice.row.${notice.id}',
              ListTile(
                title: Text(notice.title),
                onTap: () => Navigator.of(context).push(
                  MaterialPageRoute<void>(builder: (_) => NoticeDetailScreen(model: model, id: notice.id)),
                ),
              ),
            ),
        ],
      ),
    );
  }
}

/// Notice Detail, pushed by tapping a Notices row (SPEC §5.5).
class NoticeDetailScreen extends StatelessWidget {
  const NoticeDetailScreen({super.key, required this.model, required this.id});

  final AppModel model;
  final int id;

  @override
  Widget build(BuildContext context) {
    final notice = model.notice(id);
    return Scaffold(
      appBar: detailAppBar(notice?.title ?? 'Notice $id'),
      body: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            aid('notice.detail.title', Text(notice?.title ?? 'Notice $id', style: Theme.of(context).textTheme.titleLarge)),
            aid('notice.detail.body', Text(notice?.body ?? '')),
          ],
        ),
      ),
    );
  }
}
